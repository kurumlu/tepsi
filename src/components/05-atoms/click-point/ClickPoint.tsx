import * as React from 'react';
import { FunctionComponent, MouseEvent } from 'react';
import cn from 'classnames';

const CLICK_POINT_RADIUS = 16;
const CLICK_POINT_STROKE = 2;
const CLICK_POINT_PLUS_LENGTH = 10;

export type ClickPointProps = {
  x?: string | number;
  y?: string | number;
  onClick: (e?: MouseEvent) => void;
  isActive?: boolean;
};

export const ClickPoint: FunctionComponent<ClickPointProps> = ({
  x = '50%',
  y = '50%',
  onClick,
  isActive,
}) => (
  <svg
    className={cn('as-a-click-point', {
      'as-a-click-point--is-active': isActive,
    })}
    x={x}
    y={y}
    onClick={onClick}
    tabIndex={0}>
    <circle
      className="as-a-click-point__circle"
      cx="0"
      cy="0"
      r={CLICK_POINT_RADIUS}
      strokeWidth={CLICK_POINT_STROKE}
    />
    <rect
      x={-1 * (CLICK_POINT_PLUS_LENGTH / 2)}
      y={-1 * (CLICK_POINT_STROKE / 2)}
      width={CLICK_POINT_PLUS_LENGTH}
      height={CLICK_POINT_STROKE}
      strokeWidth="0"
    />
    <rect
      x={-1 * (CLICK_POINT_STROKE / 2)}
      y={-1 * (CLICK_POINT_PLUS_LENGTH / 2)}
      width={CLICK_POINT_STROKE}
      height={CLICK_POINT_PLUS_LENGTH}
      strokeWidth="0"
    />
  </svg>
);
