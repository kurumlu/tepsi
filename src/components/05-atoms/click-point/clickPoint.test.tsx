import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { ClickPoint, ClickPointProps } from './ClickPoint';
import { clickPointMock } from './clickPoint.mock';

describe('<ClickPoint />', () => {
  it('renders with default props', () => {
    const onClick = jest.fn();
    const component: ShallowWrapper<ClickPointProps> = shallow(
      <ClickPoint onClick={onClick} />
    );

    expect(component).toMatchSnapshot('default props');
  });

  it('renders with props', () => {
    const component: ShallowWrapper<ClickPointProps> = shallow(
      <ClickPoint {...clickPointMock} />
    );

    expect(component).toMatchSnapshot('with props');
  });

  it('renders as active', () => {
    const component: ShallowWrapper<ClickPointProps> = shallow(
      <ClickPoint {...clickPointMock} isActive />
    );

    expect(component.hasClass('as-a-click-point--is-active')).toBeTruthy();
  });

  it('calls onClick callback by clicking', () => {
    const onClick = jest.fn();
    const component: ShallowWrapper<ClickPointProps> = shallow(
      <ClickPoint onClick={onClick} />
    );

    component.simulate('click');
    expect(onClick).toHaveBeenCalled();
  });
});
