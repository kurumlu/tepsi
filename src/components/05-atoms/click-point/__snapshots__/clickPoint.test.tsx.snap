// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClickPoint /> renders with default props: default props 1`] = `
<svg
  className="sd-a-click-point"
  onClick={[MockFunction]}
  tabIndex={0}
  x="50%"
  y="50%"
>
  <circle
    className="sd-a-click-point__circle"
    cx="0"
    cy="0"
    r={16}
    strokeWidth={2}
  />
  <rect
    height={2}
    strokeWidth="0"
    width={10}
    x={-5}
    y={-1}
  />
  <rect
    height={10}
    strokeWidth="0"
    width={2}
    x={-1}
    y={-5}
  />
</svg>
`;

exports[`<ClickPoint /> renders with props: with props 1`] = `
<svg
  className="sd-a-click-point"
  onClick={[Function]}
  tabIndex={0}
  x="25%"
  y="40%"
>
  <circle
    className="sd-a-click-point__circle"
    cx="0"
    cy="0"
    r={16}
    strokeWidth={2}
  />
  <rect
    height={2}
    strokeWidth="0"
    width={10}
    x={-5}
    y={-1}
  />
  <rect
    height={10}
    strokeWidth="0"
    width={2}
    x={-1}
    y={-5}
  />
</svg>
`;
